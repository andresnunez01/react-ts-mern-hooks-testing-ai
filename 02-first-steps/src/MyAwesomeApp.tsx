import type { CSSProperties } from "react";

interface Props {
    firstName: string;
    lastName: string
}

const favoriteGames = ['Hollow Knight', 'Silksong', 'RDR2', 'Bioshock'];
const isActive = false;
const address = {
    zipCode: '232323',
    country: 'Mexico',
}


const myStyles: CSSProperties = {
    backgroundColor: '#fafafa',
    borderRadius: isActive ? 10 : 20,
    padding: 10,
    marginTop: 30,
}

export const MyAwesomeApp = ( { firstName, lastName }: Props ) => {

    // const firstName = 'Andres';
    // const lastName = 'Nunez'
    return <div data-testid="div-app">
        <h1 data-testid="first-name-title">
            { firstName }
        </h1>
        <h3>{ lastName }</h3>

        <p></p>
        <p className="mi-clase-favorita">{ favoriteGames.join(', ') }</p>

        <h1>{ isActive ? 'Activo' : 'No activo' }</h1>

        <p
        style={ myStyles }
        >{ JSON.stringify(address) }</p>
    </div>

}


// export function MyAwesomeApp ( { firstname, lastname }: Props ) {

//     return <>
//         <h1> Firstname: { firstname }</h1>
//         <h3> Lastname: { lastname }</h3>
//     </>

// }