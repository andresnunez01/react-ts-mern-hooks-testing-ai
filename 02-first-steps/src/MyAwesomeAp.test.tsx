import { describe, expect, test } from "vitest"
import { render, screen } from '@testing-library/react'
import { MyAwesomeApp } from "./MyAwesomeApp"

describe('MyAwesomwApp', () => {
    test('should render firstName and lastName - container', () => {

        const { container } = render(<MyAwesomeApp firstName='Andres' lastName='Nunez'></MyAwesomeApp>)
        //screen.debug();
        // console.log( container.innerHTML )


        console.log('\n\n\n\nimpresion h1')
        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');


        expect(h1?.innerHTML).toBe('Andres');
        expect(h3?.innerHTML).toContain('Nunez');


    });

    test('should render firstName and lastName - screen', () => {

        // const { container } = render(<MyAwesomeApp firstName='Andres' lastName='Nunez'></MyAwesomeApp>)
        render(<MyAwesomeApp firstName='Andres' lastName='Nunez'/>)
        //screen.debug();


        
        // const h1 = screen.getByRole('heading', {
        //     level: 1,
        // });
        const h1 = screen.getByTestId('first-name-title')

        expect(h1.innerHTML).toContain('Andres')



    });

    test('should match snapshot - container', () => {
        const { container } = render(<MyAwesomeApp firstName="Andres" lastName="Nunez"/>);

        expect(container).toMatchSnapshot();
    })

    test('should match snapshot - screen', () => {
        render(<MyAwesomeApp firstName="Andres" lastName="Nunez"/>);

        expect(screen.getByTestId('div-app')).toMatchSnapshot();
    })
})

