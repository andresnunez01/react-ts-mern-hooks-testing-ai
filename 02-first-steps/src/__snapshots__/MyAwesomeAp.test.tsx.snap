// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomwApp > should match snapshot - container 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Andres
    </h1>
    <h3>
      Nunez
    </h3>
    <p />
    <p
      class="mi-clase-favorita"
    >
      Hollow Knight, Silksong, RDR2, Bioshock
    </p>
    <h1>
      No activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"232323","country":"Mexico"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomwApp > should match snapshot - screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Andres
  </h1>
  <h3>
    Nunez
  </h3>
  <p />
  <p
    class="mi-clase-favorita"
  >
    Hollow Knight, Silksong, RDR2, Bioshock
  </p>
  <h1>
    No activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"232323","country":"Mexico"}
  </p>
</div>
`;
